python/hail/expr/functions.py

```diff
@@ -1303,6 +1303,48 @@ def log10(x) -> Float64Expression:
     return _func("log10", tfloat64, x)
 
 
+@typecheck(args=expr_any)
+def coalesce(*args):
+    """Returns the first non-missing value of `args`.
+
+    Examples
+    --------
+    >>> x1 = hl.null('int')
+    >>> x2 = 2
+    >>> hl.coalesce(x1, x2).value
+    2
+
+    Notes
+    -----
+    All arguments must have the same type, or must be of compatible types
```

Comment:
does compatible mean "convertible to a common type"? If so, I think the latter is more precise